ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js

```diff
@@ -463,6 +495,69 @@ define([
             }
         },
 
+        setBboxDmsNorth: function() {
+            this.setLatLonFromDms('dmsNorth', 'dmsNorthDirection', 'north');
+        },
+
+        setBboxDmsSouth: function() {
+            this.setLatLonFromDms('dmsSouth', 'dmsSouthDirection', 'south');
+        },
+
+        setBboxDmsEast: function() {
+            this.setLatLonFromDms('dmsEast', 'dmsEastDirection', 'east');
+        },
+
+        setBboxDmsWest: function() {
+            this.setLatLonFromDms('dmsWest', 'dmsWestDirection', 'west');
+        },
+
+        setBboxDmsFromMap: function() {
+            const dmsNorth = ddToDmsCoordinateLat(this.get('mapNorth'), getSecondsPrecision(this.get('dmsNorth')));
```

Comment:
The primary reason why this code cares about the seconds precision has to do with floating point precision. When we convert a DMS coordinate to decimal degrees and back, the seconds value is not always exactly the same because we lose some precision. When a user finishes entering a DMS coordinate, we convert the value to decimal degrees and if necessary adjust the value. Then we convert it back to DMS to display to the user again. This resulted in a bad user experience since the user could enter a DMS coordinate with an integer number of seconds (e.g. "10°10'10") but the value would immediately change to something like "10°10'09.999999999999432". Maintaining the level of precision entered by the user prevents this problem.

This is related to the comment you just added regarding `toFixed()`, which I chose not to use since it adds trailing zeros.

I added a default seconds precision of 4 so when the user draws a box the seconds aren't displayed as unnecessarily long decimals (which I think would be misleading to the user since I doubt it's possible to get beyond that degree of precision by drawing on the map anyway - see https://en.wikipedia.org/wiki/Decimal_degrees).

The "seconds precision" only affects what is _displayed_ to the user. The underlying map coordinates are always in decimal degrees which are not changed.

Reply:
I agree with your point about the precision. I think the zeros are important though. In your first screenshot, the seconds have between 1 and 4 decimal places. Judging just from the West coordinate, I'd say the map has 0.0001 second precision. But the North coordinate would imply just 0.1 second precision.

Reply:
I figured since we don't add trailing zeros anywhere else in the UI I wouldn't add them here. I also thought it would be best to leave the user's input as-is if possible. And I'd guess that a user who is that concerned about precision would adjust their coordinates manually. But if you still think we should add them then I'll do it. It's a simple change.

Reply:
I'm fine with what you have if it's consistent with the rest of the UI